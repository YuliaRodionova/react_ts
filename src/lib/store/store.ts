import { legacy_createStore, combineReducers, applyMiddleware, compose } from "redux";
// import { combineReducers, configureStore } from "@reduxjs/toolkit";
import cartReducer from "../reducers/cart";
import { loadState } from "../localStorage/localStorage";

const rootReducer = combineReducers({
    cartList: cartReducer
});

const persistedState = loadState();

// export const setupStore = () => {
//     return configureStore({
//         reducer: rootReducer
//     })
// }

const store = legacy_createStore(rootReducer, persistedState);

export default store;

export type RootState = ReturnType<typeof store.getState>;

// export type RootState = ReturnType<typeof rootReducer>
// export type AppStore = ReturnType<typeof setupStore>
// export type AppDispatch = AppStore['dispatch']