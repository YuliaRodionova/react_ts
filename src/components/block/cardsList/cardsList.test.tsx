import { render, screen } from '@testing-library/react';
import CardsList from './cardsList';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../../lib/store/store';
import * as reduxHooks from 'react-redux';

jest.mock('react-redux')

test('Card-list element: should create list with items', () => {
    const mockTestProducts = [{
        id: 7,
        img: "/productImg8.jpg",
        productName: "Мыло",
        unit: "мл",
        weight: 300,
        code: 4604049097552,
        producer: "A",
        brand: "D",
        price: 40,
    },
    {
        id: 8,
        img: "/productImg9.jpg",
        productName: "Крем",
        unit: "мл",
        weight: 45,
        code: 46040490975555,
        producer: "B",
        brand: "J",
        price: 50,
    },
    {
        id: 9,
        img: "/productImg7.jpg",
        productName: "Мочалка",
        unit: "г",
        weight: 30,
        code: 4604049097554534,
        producer: "C",
        brand: "E",
        price: 60,
    }]
    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue(mockTestProducts);

    const { container } = render(
        <BrowserRouter>
            <Provider store={store}>
                <CardsList />
            </Provider>
        </BrowserRouter>

    )

    expect(container).toMatchSnapshot();
});