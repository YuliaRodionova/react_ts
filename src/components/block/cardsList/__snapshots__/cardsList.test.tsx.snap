// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card-list element: should create list with items 1`] = `<div />`;
